app/models/web_hook.rb

```diff
@@ -101,8 +108,20 @@ def request(data)
                           })
   end
 
-  def send_payload(data, ignore_errors: false)
+  def send_payload(data, ignore_errors: false, extra_log_attrs: {})
+    start_time = Time.current
     response = request(data).run
+    StructuredLog.capture("WEB_HOOK_SENT",
+                          {
+                            webhook_id: id,
+                            response_timed_out: response.timed_out?,
+                            response_code: response.code,
+                            response_success: response.success?,
+                            # this would be in datadog already if we have a trace, but it seems we
+                            # don't always have a trace.
+                            request_duration: Time.current - start_time,
```

Comment:
There is a `total_time` field available on the response that we could use here instead of during the `Time.current` comparison.

Reply:
https://github.com/typhoeus/typhoeus/blob/f5c5751df49089da89fc2683a23df04850a45604/lib/typhoeus/response/informations.rb#L87-L97

Reply:
trying to decide if I trust it; having trouble finding the right code where it's set but I bet it is returning this https://curl.se/libcurl/c/CURLINFO_TOTAL_TIME.html

I'm wondering if typhoeus can ever do anything interesting that isn't included in that

Reply:
I dunno you're probably right here but I feel like doing the stupid thing will save me some thinking 🙃